.github/workflows/submodule-pin-check.yml

```diff
@@ -18,11 +18,32 @@ jobs:
         with:
           fetch-depth: 0
           submodules: true
+          ref: "${{ github.event.pull_request.merge_commit_sha }}"
 
       - name: Check all submodules are ancestors of origin/HEAD or configured branch
         run: |
           status_state="pending"
-          if ${{ github.workspace }}/.github/workflows/submodule-pin-check.sh; then
+          declare -Ar exceptions=(
+            [contracts]=origin/develop
+            [nitro-testnode]=origin/master
+            
+            #TODO Rachel to check these are the intended branches.
+            [arbitrator/langs/c]=origin/vm-storage-cache
+            [arbitrator/tools/wasmer]=origin/adopt-v4.2.8
+          )
+          divergent=0
+          for mod in `git submodule --quiet foreach 'echo $name'`; do
+            branch=origin/HEAD
+            if [[ -v exceptions[$mod] ]]; then
+              branch=${exceptions[$mod]}
+            fi
+            
+            if ! git -C $mod merge-base --is-ancestor HEAD $branch; then
+              echo $mod diverges from $branch
+              divergent=1
+            fi
+          done
+          if $divergent; then
```

Comment:
I believe this always fails as-is and goes to the else. I think you want `if [ $divergent -eq 1 ]; then`

![image](https://github.com/user-attachments/assets/5b797a9c-a595-4943-8b0d-5b0d6631b06e)

Reply:
To clarify why it does this, bash will turn your current expression into `if 1; then` by substituting `$divergent` for `1`, but then it'll error with `1: command not found` . Two options are either making `$divergent` either `true` or `false` which are both commands (but this is a bit hacky imo as it's executing whatever's in `$divergent`) or to use an integer comparison like I recommend in the above comment

Reply:
Thanks for catching this. Done!

Reply:
Also qq, why was this not any issue in the existing setup? Like it should failed for the script also right? we were not comparing to one then also?

Reply:
It seems like something else is the issue, it is still going to else only.....

Reply:
Actually shouldn't it compare to 0 ? Since divergent is zero when everything is alright?